### PR TITLE
Feat/exceptions log

### PR DESCRIPTION
**Pull request by StevenHee**
Display the exceptions log for connection updates made by the current user today. The records list component isn't suitable to display the exceptions due to the different data models and features like pagination, search and filtering are not required. However, it feels appropriate to add the link as a new tab under connections. A new component and service is used to fetch and display the exceptions log which is conditionally displayed in the connections section according to the filter property.

TIS21-5151: Show log of errors when adding/removing connections

### CODE REVIEW

File: src/app/connections/exceptions-log/exceptions-log.component.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    data-test="NoExceptionsMessage"
+    class="mt-20 d-flex justify-content-center"
+  >
+    No exceptions found.
+  </div>
+</ng-container>
+
+<ng-container id="ExceptionDataFound" *ngIf="pageStatus === 'ok'">
+  <h2 class="ml-20 mt-20">
+    List of errors from unsuccessful connection updates made today by currennt
```
Comment (CaiWillis) [inline src/app/connections/exceptions-log/exceptions-log.component.html:25]: "current"

Comment (CaiWillis) [reply src/app/connections/exceptions-log/exceptions-log.component.html:25]: Also just "unsuccessful connections" sounds smoother I think (I know technically these are for disconnections too but . . .)

File: src/app/connections/exceptions-log/mock-data/exceptions-log-spec-data.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { IException } from "../exceptions-log.interface";
+
+export const exceptionsJson = `
+[
+   
+    {
+        "gmcId": "1",
+        "errorMessage": "Error message",
+        "timestamp": "2023-10-18T14:45:47.337",
```
Comment (CaiWillis) [inline src/app/connections/exceptions-log/mock-data/exceptions-log-spec-data.ts:9]: Will these times work in future? (I'm guessing the "flush" method used just returns this no matter what day it is but just checking)

Comment (StevenHee) [reply src/app/connections/exceptions-log/mock-data/exceptions-log-spec-data.ts:9]: Yeah, this uses the test data so should always pass.

**Review by CaiWillis — APPROVED** [review]

**Pull request merged**